### ISSUE TITLE
Potential metrics to consider

### ISSUE THREAD

**Issue by Popolechien** [offspot/metrics#5]
Here some feedback from Orange as to which metrics they'd be interested in:

- Weekly/monthly duration of operation for the device;
- Weekly/monthly duration of use for each zim file; 
- Number of files added/deleted (via EduPi presumably)

**Comment by kelson42** [comment]
Naive proposal: Time duration of consecutive time periods (depending of the time scale granularity needed) with "activity".

**Comment by benoit74** [comment]
I'm sorry but I don't get it.
Let's work on a naive example, I like it. Let's say we want the weekly and monthly duration of use of a given Zim. During this month, there has been activity on this Zim during three days: Monday 7 at 8:01, 8:02, 8:03, 10:31, Thursday 10 at 9:05 and Monday 14 at. 10:03, 10:04, 14:03, 14:06 and 18:04.
What is monthly duration of use ? What is the weekly duration for the week from 7 to 13 ? For the week from 14 to 21 ?

**Comment by Popolechien** [comment]
Yeah I see the issue : we cannot monitor client-side how long a page that has already been requested has actually been opened/read.
How about then the number of page requests over a given time period? 

In the odd chance that the WMF has research on average time spent reading an entry (much like they know that 60% of readers stop at the introduction), could we assume then the overall time spent?

**Comment by kelson42** [comment]
@benoit74 I will explain it to you directly, but think about the time granularity (missing on your example) and the rest should be simple.

**Comment by benoit74** [comment]
@Popolechien 
Yes we can assume an average time spent per resource, but I would rather report the "raw" KPI about the number of resources consumed, and let the user do the conversion if they see fit, to present only "real" data and let the interpretation for those who want to do it, so that anyone can choose how to do this (because the time spent reading a wikipedia page is probably very different from the one spent on a TEDx video, ...).

We could monitor it client-side, but it is probably rather complex (especially since we have many kinds of clients, and they probably do not all support the same kind of system to report this) and for now the decision has been made to base metrics on information available on the offspot. There would be probably also a lot of bias / noise in a client side measure (cases where someone opened the page, left it open but never read anything, cases where someone downloaded an ePub and read it disconnected from the offspot, ...)

@kelson42 ok ...

**Comment by rgaudin** [comment]
- 1. is preferably handled by the metrics container.
- 3. could probably be handled by mounting/listing the EduPi folder but it can be an opportunity to setup app-specific data _transmission_.
- 2. Awaiting @kelson42's insight but at the very least we know that _one minute_ is our lowest granularity. We can use a larger window (2mn? 5mn?) as the default _duration of activity after HTTP requests stopped_. There are a number of cases that won't fit this scenario but those are user-understandable I believe. Note that most videos use range requests in ZIM at least so would be less impacted. Downloads are arguably not activity. The most valid _longer_ case I can think of is the long reading of WP-like content but there's not much we can do about it.

**Comment by benoit74** [comment]
@Popolechien : do you agree that everything in this issue has been transferred to #8 so I can close this issue ?

**Comment by Popolechien** [comment]
Yup
